_episodes/11.md

```diff
@@ -0,0 +1,258 @@
+or instance. The foreign keys connect these big fact tables to smaller dimension
+tables that when joined provide human readable context to enrich the recordings
+in the fact table. The schema ends up looking like a star with the fact table at
+the center. In essence, you just need to remember when someone is describing a
+fact table they are saying it is a bigger table that is likely going to end up
+on the probe side of a join, where a dimension is more likely a candidate to fit
+into memory on the build side of a join.
+
+So let's get onto the Dynamic Filtering shall we? First,let's cover a few
+concepts about Dynamic Filtering, then cover some variations of this concept.
+
+Dynamic Filtering takes advantage of joins with big fact tables to smaller
+dimension tables. What makes this filtering different from other types of
+filtering is that you are using the smaller build table that is loaded at query
+time to generate a list of values that exist in the join column between the
+build table and probe table. We know that only values that match these criteria
+are going to be returned from the sales side, so we can use this dynamically
+generated list as a pushdown predicate on the join column of the probe side.
+This means we are still scanning this data but only sending the subset that will
+answer the query. We can look at the blog written for the original local dynamic
+filtering implementation by Roman Zeyde for more insights on the original
+implementation for Dynamic Filtering before Raunaq's changes.
+
+<https://trino.io/blog/2019/06/30/dynamic-filtering.html>
+
+Local Dynamic Filtering is definitely beneficial, in that it generates a list
+of values that can be used to avoid sending some unnecessary rows, but what if 
+we could cut out entire splits of unnecessary partitions from the query
+execution? In the next iteration of Dynamic Filtering, called Dynamic Partition 
+Pruning, we can take advantage of probe tables that are partitioned and use that
+information to tell the coordinator to prune or eliminate the splits that do not
+need to be scheduled. This only works, however, if the query contains a
+predicate on the column that the partition exists on. You can require that these
+predicates exist in all queries. If you want to enforce this for dynamic
+filtering by setting this property.
+
+{% highlight text %}hive.query-partition-filter-required=true
+{% endhighlight %}
+
+With that basic understanding, let's move on to the PR that implemented Dynamic
+Partition Pruning!
+
+## PR of the week: PR 1072 Implement Dynamic Partition Pruning
+
+In this week's pull request <https://github.com/trinodb/trino/pull/1072> we
+return with Raunaq Morarka and Karol Sobczak. This PR effectively brings in the 
+second iteration of Dynamic Filtering, dynamic partition pruning, where instead
+of relying on local dynamic filtering we involved the coordinator to update the
+query plan and prune out extra splits that wouldn't be needed based on the
```

Comment:
we don't update the query plan in coordinator, we collect dynamic filters from the workers there and use them when they're ready

Reply:
ah right, I forgot to change this..you corrected that during the show. Fixing.

Reply:
Fixed